notifications_utils/template.py

```diff
@@ -388,7 +390,7 @@ def __str__(self):
 
         return self.jinja_template.render(
             {
-                "body": get_html_email_body(self.content, self.values),
+                "body": get_html_email_body(self.content, self.values, html="passthrough" if self.allow_html else "escape"),
```

Comment:
If this passes the HTML through, what happens if I accidentally put invalid HTML into the template?

Reply:
I will add some validation - if the html added to the template is valid, then use the "passthrough" option, else use the "escape" option.

Reply:
We probably need more then just validation. If someone gets access to notify that shouldn't, can I craft a template with a script alert xss in it? Or various other forms of attack?

We'd need to both validate the HTML is correct, plus put in a white list of allowed HTML tokens and have anything else get encoded.

Reply:
Hi Dan, these concerns are quite valid and I share these as well. 

We consider this feature for pressing need, i.e. we want to enable it for the IRCC notifications service **only** which deal with the Afghanistan situation. It's certainly a feature that requires more care especially if we expand for other services. It's not what we plan to do right now: the goal is to answer quickly to the urgent ask from IRCC, which could impact lives of some Canadians stuck in there.

To prevent other services to use this, we will have a filter on the celery task that invokes this logic and sends email. Only a predefined list of service IDs could invoke this (a list of 1 service, i.e. the IRCC notification service that deals with the Afghanistan notifications). 

If we were to make this a feature usable by other services, I really want to have HTML validation and subset of HTML tokens as you mentioned -- no less for sure. It's something to discuss within the team if we want to expand it to other services though because custom HTML is hot potato. My personal take is that there is a 1-2% use case where that could be super beneficial (like here), but we never went forward with it because of these concerns and yet, here is a situation where we could have delivered a better service to our user and faster response. I would like for Notify to have this better supported in the future if we go this route, but with a few caveats: 1- not publicized (no doc), 2- enabled only by admins on specific templates, 3- only for specific clients with special needs. 

Does that make sense? Would you have a different recommendation, given the situation with IRCC and Afghanistan?